### PR TITLE
Create authorized request controller

### PR DESCRIPTION
**Pull request by arosty**
Redid the controller with all its basic actions.

Did some basic views, nothing to care about.

### CODE REVIEW

File: app/controllers/requests_controller.rb [hunk 1]
```diff
@@ -1,33 +1,49 @@
 class RequestsController < ApplicationController
   before_action :find_request, only: [:show, :update, :destroy]
+
   def index
+    @requests = policy_scope(Request).order(created_at: :desc)
```
Comment (davidmetta) [inline app/controllers/requests_controller.rb]: alternatively we can include skip_policy_scope and just define the index method, also we would leave the Scope class in the requestPolicy in its original form.

File: app/controllers/requests_controller.rb [hunk 2]
```diff
@@ -1,33 +1,49 @@
...
   end
 
   def show
+    authorize @request
```
Comment (davidmetta) [inline app/controllers/requests_controller.rb]: alternatively we can include skip_authorization and just define the show method, also we would would not need to create the show? Class in the requestPolicy.

File: app/controllers/requests_controller.rb (hunk 1 above)
Comment (arosty) [reply app/controllers/requests_controller.rb]: Yeah seems to make sense. But how would the could look then exactly?

File: app/controllers/requests_controller.rb (hunk 2 above)
Comment (arosty) [reply app/controllers/requests_controller.rb]: How would the code look then exactly?

File: app/controllers/requests_controller.rb [hunk 3]
```diff
@@ -1,33 +1,50 @@
 class RequestsController < ApplicationController
   before_action :find_request, only: [:show, :update, :destroy]
+
   def index
+    skip_policy_scope
```
Comment (davidmetta) [inline app/controllers/requests_controller.rb]: line 5 should be under line 6

**Pull request merged**